refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
+                             format(REFGENIE_BY_CFG[str(config["config_version"])], REFGENIE_BY_CFG[str(REQ_CFG_VERSION)]))
+                return missing_digest
+        # reformat config file
+
+        def format_config(config, target_version):
+            # reformat the config file
+            for k, v in config["genomes"].items():
+                # create "aliases" section
+                v.aliases = [k]
+                # get genome digest from the server
+                response = requests.get(
+                    "http://refgenomes.databio.org:82/v3/alias/genome_digest/"+k)
+                if response.json()['detail']:
+                    # if genome asset not exist on serer, generate the digest using the local fasta asset
+                    ssc = SeqColClient({})
+                    try:
+                        d, _ = ssc.load_fasta(config["genome_folder"]+'/'+k +
+                                              '/fasta/default/'+k+'.fa')
+                    except OSError:
+                        del config["genomes"][k]
+                        continue
+                else:
+                    d = response.json()
+                # convert seek keys, childran/parent asset keys from aliases to genome digests
+                for asset_k, asset_v in v["assets"].items():
+                    for seek_k, seek_v in asset_v["tags"]["default"]["seek_keys"].items():
+                        asset_v["tags"]["default"]["seek_keys"][seek_k] = seek_v.replace(
+                            k, d)
+
+                    if (asset_k == "fasta" and "asset_children" in asset_v["tags"]["default"]):
+                        for i in range(len(asset_v["tags"]["default"]["asset_children"])):
+                            asset_v["tags"]["default"]["asset_children"][i] = asset_v["tags"]["default"]["asset_children"][i].replace(
+                                k, d)
+                    elif asset_k != "fasta":
+                        for i in range(len(asset_v["tags"]["default"]["asset_parents"])):
+                            asset_v["tags"]["default"]["asset_parents"][i] = asset_v["tags"]["default"]["asset_parents"][i].replace(
+                                k, d)
+                # use the genome digest as primary keys
+                config["genomes"][d] = config["genomes"].pop(k)
+                # remove old "genome_digest" section
+                del config["genomes"][d]["genome_digest"]
+                # change the config_version
+                config["config_version"] = target_version
+                # write over the config file
+                config.write()
+            return config
+        # restructure the genome_folder
+
+        def alter_file_tree(config):
+            my_genome = {}
+            for k, v in config["genomes"].items():
+                my_genome.update([(v["aliases"][0], k)])
+            os.mkdir(config["genome_folder"]+'/data/')
+            os.mkdir(config["genome_folder"]+'/alias/')
```

Comment:
instead of stitching the paths together use the `RefGenConf` class properties I've defined: 
https://github.com/refgenie/refgenconf/blob/a4ec5c29fa3d286bbbd8a86e959b2e42fc84006f/refgenconf/refgenconf.py#L193-L209